libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java

```diff
@@ -52,93 +52,135 @@
         return getDecryptingCipher(keyBytes, iv);
     }
 
     private static Cipher getDecryptingCipher(byte[] keyBytes, byte[] iv)
-        throws InvalidAlgorithmParameterException, InvalidKeyException {
-        final Cipher cipher = getBestCipher();
+            throws InvalidAlgorithmParameterException, InvalidKeyException {
+        final Cipher cipher = getBestCipher(AES_GCM_CIPHER);
+        final SecretKeySpec skeySpec = new SecretKeySpec(keyBytes, cipher.getAlgorithm());
+        final IvParameterSpec ivSpec = new IvParameterSpec(iv);
+        cipher.init(Cipher.DECRYPT_MODE, skeySpec, ivSpec);
+        cipher.updateAAD(ivSpec.getIV());
+        return cipher;
+    }
+
+    /*
+     * TODO: Remove this in Mobile SDK 10.0.
+     */
+    private static Cipher getLegacyDecryptingCipher(byte[] keyBytes, byte[] iv)
+            throws InvalidAlgorithmParameterException, InvalidKeyException {
+        final Cipher cipher = getBestCipher(AES_CBC_CIPHER);
         final SecretKeySpec skeySpec = new SecretKeySpec(keyBytes, cipher.getAlgorithm());
         final IvParameterSpec ivSpec = new IvParameterSpec(iv);
         cipher.init(Cipher.DECRYPT_MODE, skeySpec, ivSpec);
         return cipher;
     }
 
     /**
-     * Decrypts data with key using AES-128.
+     * Decrypts data with key using AES/CBC/PKCS5Padding. Should be used only for
+     * migration of encrypted data from an older version to Mobile SDK 8.2.
      *
      * @param data Data.
      * @param key Base64 encoded 128 bit key or null (to leave data unchanged).
      * @return Decrypted data.
+     * @deprecated Will be removed in Mobile SDK 10.0.
      */
-    public static String decrypt(String data, String key) {
+    public static String legacyDecrypt(String data, String key) {
         return decrypt(data, key, new byte[16]);
     }
 
     /**
-     * Decrypts data with key using AES-128.
+     * Decrypts data with key using AES/GCM/NoPadding.
      *
      * @param data Data.
      * @param key Base64 encoded 128 bit key or null (to leave data unchanged).
-     * @param iv Initialization vector.
      * @return Decrypted data.
      */
-    public static String decrypt(String data, String key, byte[] iv) {
-        if (TextUtils.isEmpty(key) || data == null) {
-            return data;
-        }
-        return decrypt(data.getBytes(), key, iv);
+    public static String decrypt(String data, String key) {
+        return decrypt(data, key, new byte[12]);
     }
 
     /**
-     * Decrypts data with key using AES-128.
+     * Decrypts data with key using AES/CBC/PKCS5Padding. Should be used only for
+     * migration of encrypted data from an older version to Mobile SDK 8.2.
      *
      * @param data Data.
      * @param key Key.
      * @return Decrypted data.
+     * @deprecated Will be removed in Mobile SDK 10.0.
      */
-    public static String decrypt(byte[] data, String key) {
+    public static String legacyDecrypt(byte[] data, String key) {
```

Comment:
Could you add a legacyEncrypt (could be package level)? That will make writing upgrade tests easier.